[5.2] Add chunkById query builder tests

kevindoole opened the pull request:
This just adds a test for query builder chunkById, and fixes up some crappy docblocks from my last pr.

There were no existing tests for the query builder chunk method -- possibly because the method is essentially duped from the eloquent builder file. Seems nice to have both sides tested.

Happy to add a couple more tests if they bring any value.

kevindoole committed:
Fix docblock formats

kevindoole committed:
Add query builder chunkById test; slight method cleanup

kevindoole committed:
Revert a param name

kevindoole committed:
Adjust whitespace in one more docblock

vlakoff commented on src/Illuminate/Database/Eloquent/Builder.php:
```diff
@@ -330,8 +330,8 @@ public function value($column)
      *
-     * @param  int  $count
-     * @param  callable  $callback
+     * @param  int      $count
```
You'll get hit on the fingers for these! :scream:

The formats are `@param__type__$variable` and `@return_type`.
Always the same number of spaces, no vertical alignment.

kevindoole committed:
Adjust various code style infringements

kevindoole commented:
Ha... whoops. I started that way, but then realized that other methods in the file seemed to vertically align, so i changed it. Changed it back now :smile: 

Thanks for clarifying.

vlakoff commented:
Some phpdocs don't follow this convention simply because they haven't been updated. It's not supported by StyleCI ;)

mark86092 commented:
IMO, other part of phpdocs fix should go to new PR

vlakoff commented on src/Illuminate/Database/Eloquent/Builder.php:
```diff
@@ -332,7 +332,7 @@ public function value($column)
      * @param  int  $count
      * @param  callable  $callback
-     * @return bool
+     * @return  bool
```
I wrote `@return_type`, one space!

(also you may want to stash the commits to not clutter git history)

GrahamCampbell replied:
don't worry about phpdoc cs - as long a it's valid phpdoc, taylor or I can clean it up after merge

GrahamCampbell replied:
Please do squash this to one commit though.

vlakoff commented:
More important, because of the mocking, these tests failed to detect that the results contains only the id column, and I also noted in https://github.com/laravel/framework/pull/12861#issuecomment-202780961.

vlakoff commented:
No thoughts concerning the "only id column in results" issue? Seems pretty important to me.

vlakoff commented:
@kevindoole Any news on this "select only id" stuff? As a reminder, this goes unnoticed because tests mock `pageAfterId`.

kevindoole commented:
Yeah, I've been dragging my heels with this! I'll get it done tonight; looks like probably a change to one line, unless I figure out a nicer way to test it.

kevindoole commented:
@vlakoff Looks like someone beat me to it :+1: 
I'm gonna still take a quick look at some of the duplication there.

vlakoff commented:
> Looks like someone beat me to it :+1:

Yep: #13137
